### ISSUE TITLE
[Question] - How is this project maintained ?

### ISSUE THREAD

**Issue by mrik974** [corazawaf/coraza-caddy#179]
Hi. I came across this project while gathering information about Coraza. It seems that it's, so far, the most advanced implementation of Coraza in a reverse proxy/webserver (correct me if I'm wrong about it). 
It brings me to some questions, regarding the state of this repository.

1 - The latest release is more than 40 commits behind the `main` branch.
When we run `xcaddy build --with github.com/corazawaf/coraza-caddy/v2`, which version/commit/tag are we running ? 

2 - The version 2 is still in Release Candidate, but this was a year ago. Do you think it will change state after 1 year of commits ? 

3 - It seems that the integration of the latest versions of CRS have failed, a month ago. Does it mean there are issues with the maintenance ?

I know, as open source software users, we're clearly not entitled to anything, and are not in a position to demand anything either. 
I'm just curious about the state and maintenance of this cool project :). I see that you are active, answering to people in issues, but some parts are not really moving and I don't understand why.

**Comment by jcchavezs** [comment]
Hi @mrik974 thanks for coming by. Indeed this is an important connector but since we are all volunteers we dedicate our efforts to what people request the most. In this case, caddy is hopping for someone to use it and help to triage bugs.  I will try to dedicate some time to it this week and see the current issues.

**Comment by skrlance** [comment]
In the difference of a month release candidates were released 1 year 3 months ago and there is no sign of releasing stable version since then every week I check if stable version has been released but not yet!

**Comment by fzipi** [comment]
Did you read the previous comment, being this a volunteer project? Do you have time to donate to move the needle forward towards a "stable release"?

**Comment by skrlance** [comment]
> Did you read the previous comment, being this a volunteer project? Do you have time to donate to move the needle forward towards a "stable release"?

Sorry I was just expressing that I love this awesome project and very curious if there will be an update!

**Comment by jptosso** [comment]
There will beThe problem is that we do not get technical feedback from our users and it takes too much time to debug issues with words like “this is not working”.This project has a high priority for us.We should push a tag updating coraza version On Nov 28, 2024, at 15:40, Shekhar ***@***.***> wrote:﻿

Did you read the previous comment, being this a volunteer project? Do you have time to donate to move the needle forward towards a "stable release"?

Sorry I was just expressing that I love this awesome project and very curious if there will be an update!

**Comment by jcchavezs** [comment]
@skrlance the path towards stable release is people using RCs, reporting bugs or ideally fixing them.

**Comment by skrlance** [comment]
> @skrlance the path towards stable release is people using RCs, reporting bugs or ideally fixing them.

Yes I tried to but v2 URL gives 404 error so I cannot add the RC package:

`github.com/corazawaf/coraza-caddy/v2`

**Comment by louis-lau** [comment]
@skrlance It's not an http url, it's a go module path, which uses [semantic versioning](https://semver.org/):
https://go.dev/doc/modules/version-numbers

So v2 tells golang to look at the tags, and pick the latest that matches the version according to semver:
https://github.com/corazawaf/coraza-caddy/tree/v2.0.0-rc.3

**Comment by skrlance** [comment]
> @skrlance It's not an http url, it's a go module path, which uses [semantic versioning](https://semver.org/): https://go.dev/doc/modules/version-numbers
> 
> I'm not very familiar with how golang resolves this path, but it's not just a simple http url.
> 
> I'd suggest actually using the given command to try adding it to caddy, instead of manually browsing to a module path.

Thank you for the info! Also, I have 1 final question that is it possible to compile the mainline version through xcaddy? Please provide me the command if possible!

**Comment by louis-lau** [comment]
I was only able to do it by pinning the commit. Not sure if that's actually the only way, but it's the only thing that worked for me and the only thing I found after looking at the golang docs.

So for the currently latest commit:
```
xcaddy build --with github.com/corazawaf/coraza-caddy/v2@2502703e84440c48f4c044e5a88c0c036315fd40
```

**Comment by jcchavezs** [comment]
I will launch a stable version based on 3.3.1 soon.

**Comment by jcchavezs** [comment]
We just cut 2.0.0 :)
